redbot/cogs/audio/apis.py

```diff
@@ -32,129 +17,33 @@
-                            id TEXT,
-                            type TEXT,
-                            uri TEXT,
-                            track_name TEXT,
-                            artist_name TEXT, 
-                            song_url TEXT,
-                            track_info TEXT,
-                            last_updated TEXT,
-                            last_fetched TEXT
-                        );
-                    """
-
-_INSERT_SPOTIFY_TABLE = """
-        INSERT OR REPLACE INTO 
-        spotify(id, type, uri, track_name, artist_name, 
-        song_url, track_info, last_updated, last_fetched) 
-        VALUES (:id, :type, :uri, :track_name, :artist_name, 
-        :song_url, :track_info, :last_updated, :last_fetched);
-    """
-_QUERY_SPOTIFY_TABLE = "SELECT * FROM spotify WHERE uri=:uri;"
-_UPDATE_SPOTIFY_TABLE = """UPDATE spotify
-              SET last_fetched=:last_fetched 
-              WHERE uri=:uri;"""
-
-_DROP_LAVALINK_TABLE = "DROP TABLE lavalink;"
-
-_CREATE_LAVALINK_TABLE = """
-                CREATE TABLE IF NOT EXISTS lavalink(
-                    query TEXT,
-                    data BLOB,
-                    last_updated TEXT,
-                    last_fetched TEXT
-
-                );
-            """
-
-_CREATE_UNIQUE_INDEX_LAVALINK_TABLE = (
-    "CREATE UNIQUE INDEX IF NOT EXISTS idx_lavalink_query ON lavalink (query);"
-)
-
-_INSERT_LAVALINK_TABLE = """
-        INSERT OR REPLACE INTO 
-        lavalink(query,  data, last_updated, last_fetched) 
-        VALUES (:query, :data, :last_updated, :last_fetched);
-    """
-_QUERY_LAVALINK_TABLE = "SELECT * FROM lavalink WHERE query=:query;"
-_QUERY_LAST_FETCHED_LAVALINK_TABLE = (
-    "SELECT * FROM lavalink "
-    "WHERE last_fetched LIKE :day1"
-    " OR last_fetched LIKE :day2"
-    " OR last_fetched LIKE :day3"
-    " OR last_fetched LIKE :day4"
-    " OR last_fetched LIKE :day5"
-    " OR last_fetched LIKE :day6"
-    " OR last_fetched LIKE :day7;"
-)
-_UPDATE_LAVALINK_TABLE = """UPDATE lavalink
-              SET last_fetched=:last_fetched 
-              WHERE query=:query;"""
-
-_PARSER = {
-    "youtube": {
-        "insert": _INSERT_YOUTUBE_TABLE,
-        "youtube_url": {"query": _QUERY_YOUTUBE_TABLE},
-        "update": _UPDATE_YOUTUBE_TABLE,
-    },
-    "spotify": {
-        "insert": _INSERT_SPOTIFY_TABLE,
-        "track_info": {"query": _QUERY_SPOTIFY_TABLE},
-        "update": _UPDATE_SPOTIFY_TABLE,
-    },
-    "lavalink": {
-        "insert": _INSERT_LAVALINK_TABLE,
-        "data": {"query": _QUERY_LAVALINK_TABLE, "played": _QUERY_LAST_FETCHED_LAVALINK_TABLE},
-        "update": _UPDATE_LAVALINK_TABLE,
-    },
-}
-
 _TOP_100_GLOBALS = "https://www.youtube.com/playlist?list=PL4fGSI1pDJn6puJdseH2Rt9sMvt9E2M4i"
 _TOP_100_US = "https://www.youtube.com/playlist?list=PL4fGSI1pDJn5rWitrRWFKdm-ulaFiIyoK"
 
+_database: CacheInterface = None
+_bot: Red = None
+_config: Config = None
```

Comment:
These type hints should be `Optional[...]`

Reply:
I know the IDE was complaining, but should they are initialised before anything in the module is used and I came across some weirdness with pycharm where when I done optional because it’s set as none here it would consider the value of this always none when it came to code completion so I though I was messing up the typehints

Reply:
Optional is correct here, though ideally during refactoring planned for #3226 we get rid of all module globals anyhow.